packages/devtools/lib/src/tables.dart

```diff
@@ -659,14 +660,44 @@ class TreeTable<T extends TreeNode<T>> extends Table<T> {
         }
         insertIndex++;
       }
-      node.isExpanded = true;
+      node.expand();
     }
 
     expand(dataObject);
 
     _selectedObject ??= dataObject;
     setRows(data);
   }
+
+  void expandAll() {
+    // Store visited nodes so that we do not expand the same root multiple
+    // times.
+    final Set<T> visited = {};
+    for (T dataObject in data) {
+      final root = dataObject.root;
+      if (!visited.contains(root)) {
+        root.expandCascading();
+        visited.add(root);
+      }
+    }
+
+    setRows(data);
+  }
+
+  void collapseAll() {
+    // Store visited nodes so that we do not expand the same root multiple
+    // times.
+    final Set<T> visited = {};
```

Comment:
nit:
it looks a bit strange at first  but the recommended way to write this is
final visited = <T>{};
The single type tells dart it is a Set not a Map.

Reply:
Without the Set<T> type, the analyzer gives errors for `contains` and `add`, thinking that it is a Map.

Reply:
Jacob's comment needed backticks - the `<T>` was there but hidden by markdown...